File: Abundant.pl
#!perl.exe
use strict;
use warnings;
use ntheory qw/divisors /;
use feature qw/say/;

=pod

=head3 NAME

Challenge 171 Abundant Number

=head3 AUTHOR

Robert DiCicco

DATE: 2022-06-28

=head3 DESCRIPTION

A number n for which the sum of divisors > 2n, or, equivalently, 
the sum of proper divisors (or aliquot sum) s(n) > n.

For example, 945 is the first Abundant Odd Number.

=cut

# AUTHOR: Robert DiCicco
# DATE: 2022-06-28
# Challenge 171 Abundant Number ( Perl )

my $n = 1;                                # Starting Number
my $count = 0;                            

while ($count < 20) {                     # Looking for 20 odd numbers
  my $sum = 0;                            # Hold our sum of divisors
  my @arr = divisors($n);
  pop(@arr);
  foreach(@arr) {                          # Add each divisor to the sum
    my $val = $_;
    $sum += $val;
  }
  if (($sum > $n) && ($n % 2 == 1)) {     # if the sum of divisors is greater than the source number,
    print "$n ";                          # and it is odd, print it
    $count++;
  }
  $n++;                                   # get next number
}
print "\n";

#---------------------------------------------------------------------
# SAMPLE OUTPUT
# 945 1575 2205 2835 3465 4095 4725 5355 5775 5985 6435 6615 6825 7245 7425 7875 8085 8415 8505 8925 

File: Abundant.jl
#!julia.exe

# AUTHOR: Robert DiCicco
# DATE: 2022-06-28
# Challenge 171 Abundant Numbers ( Julia )

function divisors(n) 
  temp = []
  for i in 1:n
    if (( n % i == 0 ) && ( n != i))
      push!(temp,i)
    end
  end
  return temp
end

n = 1
count = 0

while count < 20
  sum = 0
  arr = divisors(n)
  sz = size(arr,1)
  for j in 1:sz
    sum += arr[j]
  end
  if ((sum > n) && ( n % 2 == 1))
    print(n," ")
    global count += 1
  end
  global n += 1
end
println(" ")

#-----------------------------------------------------
# SAMPLE OUTPUT
# 945 1575 2205 2835 3465 4095 4725 5355 5775 5985 6435 6615 6825 7245 7425 7875 8085 8415 8505 8925